File: car-parts/src/main/java/org/jesperancinha/smtd/carparts/model/jpa/Part.java
package org.jesperancinha.smtd.carparts.model.jpa;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Part {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String name;
}

File: car-parts-kotlin/src/main/kotlin/org/jesperancinha/smtd/carparts/model/jpa/Part.kt
package org.jesperancinha.smtd.carparts.model.jpa

import jakarta.persistence.*

@Entity
@Table(name = "PARTS")
data class Part(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: Long? = null,
    val name: String
)
